src/lib/zf/log.c

```diff
@@ -135,14 +136,25 @@ void zf_log(struct zf_stack* st, const char* fmt, ...)
   va_end(va);
 }
 
+void zf_log_set_file_name(const char *file)
+{
+  strncpy(zf_log_file_name, file, sizeof(zf_log_file_name));
+  zf_log_file_name[ZF_LOG_FILE_NAME_SIZE-1] = '\0';
+
+  if( strnlen(file, ZF_LOG_FILE_NAME_SIZE) > ZF_LOG_FILE_NAME_SIZE-1 ) 
```

Comment:
I would check the length is valid before trying to open or set name and error otherwise.  Don't think we should write to some truncated path that might not even be intended folder

Reply:
Agreed. Now it checks the length is valid before trying to open/set name and errors if not valid.